python/pyarrow/_plasma.pyx

```diff
@@ -487,24 +491,36 @@ cdef class PlasmaClient:
 
         Returns
         -------
-        list of buffer objects or a buffer object
-            Python buffer or list of Python buffer for the data
-            associated with the object_ids and ObjectNotAvailable if the
-            object was not available.
+        list of tuples of metadata bytes and a buffer object
+            Python tuple or list of tuples of metadata bytes and a buffer
+            object associated with the object_ids and ObjectNotAvailable if
+            the object was not available.
         """
         cdef c_vector[CObjectBuffer] object_buffers
         if isinstance(object_ids, collections.Sequence):
             results = []
             self._get_object_buffers(object_ids, timeout_ms, &object_buffers)
             for i in range(object_buffers.size()):
                 if object_buffers[i].data.get() != nullptr:
-                    size = object_buffers[i].data.get().size()
-                    results.append(object_buffers[i].data.get().data()[:size])
+                    data = pyarrow_wrap_buffer(object_buffers[i].data)
                 else:
-                    results.append(None)
+                    data = ObjectNotAvailable
+                if object_buffers[i].data.get() != nullptr:
+                    metadata = object_buffers[i].metadata.get().data()[:]
+                else:
+                    metadata = None
+                results.append((metadata, data))
             return results
         else:
-            return self.get_buffer([object_ids], timeout_ms)[0]
+            return self.get_raw_buffer([object_ids], timeout_ms)[0]
+
+    def deserialize_buffer(self, buffer, serialization_context=None):
+        return pyarrow.deserialize(buffer, serialization_context)
+
+    def get_bytes_from_buffer(self, buffer):
```

Comment:
Why do we need this method?

Reply:
The returned data buffer is the result of `pyarrow_wrap_buffer`. The buffer is easy to be used to deserialize. This function is used to returned the copied `bytes`. You can check the test function `deserialize_or_output`. Is there a better way?

Reply:
`get_raw_buffer` returns the same kind of buffer that `get_buffers` returns, right? And we never needed the helper function before, so it seems like it should not be necessary now, right?

Reply:
It is a bit different. `get_buffers` only returns the data part and throw away the metadata part. `get_raw_buffer` returns both part. We can merge the two functions.

Reply:
I guess the two probably should be merged (though that doesn't necessarily need to be in this PR (up to you)).

However, the utility function for reading the buffer still does not seem like it should be part of the plasma client.

Reply:
Yes, I will remove `deserialize_buffer `.  Do you also think that `get_bytes_from_buffer` should not be part of plasma client?

Reply:
Yeah, `get_bytes_from_buffer` is currently only used in the test, right? I'd just put that logic in the test.

It doesn't really have anything to do with plasma, it's just for buffers.

Reply:
`from pyarrow.lib cimport pyarrow_unwrap_buffer` cannot be used in python code. It is only accepted in cython code. And we may use it in Ray.

Reply:
@robertnishihara I found a function of `pyarrow.Buffer` named `to_pybytes`. Then I can remove this function.

Reply:
We can always call `buf.to_pybytes()` on the buffer to get the bytes. Does that work?

Reply:
Ah, just saw your comment. Great!